File: app/src/main/java/com/pysiak/simple/clock/ui/ClockPresenter.java
package com.pysiak.simple.clock.ui;

import android.content.Context;
import android.support.v4.app.Fragment;
import com.pysiak.simple.clock.R;

import java.util.ArrayList;
import java.util.List;

/**
 * Presenter for my main clock screen with {@link SimpleClockView}
 *
 * @author Yaroslav P.
 */
class ClockPresenter {

    private SimpleClockView simpleClockView;
    private Context context;

    ClockPresenter(Context context) {
        this.context = context;
    }

    /**
     * Attach instance of {@link SimpleClockView} to presenter
     *
     * @param simpleClockView instance of {@link SimpleClockView}
     */
    void attachView(SimpleClockView simpleClockView) {
        this.simpleClockView = simpleClockView;
    }

    /**
     * This is a method for creating new fragments and installed on PagerAdapter
     */
    void loadData() {
        List<Fragment> fragmentList = new ArrayList<>();
        List<String> titleList = new ArrayList<>();

        fragmentList.add(new TabFragment());
        fragmentList.add(new TabFragment());
        fragmentList.add(new TabFragment());
        titleList.add(context.getString(R.string.tab1));
        titleList.add(context.getString(R.string.tab2));
        titleList.add(context.getString(R.string.tab3));

        if (simpleClockView != null) {
            simpleClockView.setPagerAdapter(fragmentList, titleList);
        }
    }
}

File: app/src/main/java/com/pysiak/simple/clock/presenter/ClockPresenter.kt
package com.pysiak.simple.clock.presenter

import android.content.Context
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import com.pysiak.simple.clock.R
import com.pysiak.simple.clock.ui.SimpleClockActivity
import java.util.ArrayList


class ClockPresenter (val context: Context){

    private var simpleClockView: SimpleClockView? = null


    fun attachView(simpleClockView: SimpleClockView) {
        this.simpleClockView = simpleClockView
    }
     fun loadData() {
         val fragmentList = ArrayList<Fragment>()
         val stringList = ArrayList<String>()
         fragmentList.add(TabFragment())
         fragmentList.add(TabFragment())
         fragmentList.add(TabFragment())
         stringList.add(context.getString(R.string.tab1))
         stringList.add(context.getString(R.string.tab2))
         stringList.add(context.getString(R.string.tab3))


         simpleClockView?.setPagerAdapter(fragmentList,stringList)
    }

}












/*private val clockPresenter: List<Fragment>

    get() {
        val fList = ArrayList<Fragment>()
        fList.add(TabFragment().newInstance("Tab1"))
        fList.add(TabFragment().newInstance("Tab2"))
        fList.add(TabFragment().newInstance("Tab3"))
        return fList
    }*/